rpm/sailfishos-chum-gui.spec

```diff
@@ -58,44 +55,50 @@ Icon: https://raw.githubusercontent.com/sailfishos-chum/sailfishos-chum-gui/main
 %setup -q -n %{name}-%{version}
 
 %build
-
 # SFOS RPM cmake macro disables RPATH
-%cmake -DCHUMGUI_VERSION=%(echo %{version} | grep -Eo '^[0-9]+(\.[0-9]+)*') \
-      -DCMAKE_SKIP_RPATH:BOOL=OFF \
-      -DCMAKE_INSTALL_RPATH=%{_datadir}/%{name}/lib: \
-      -DGITHUB_TOKEN=%(cat %{SOURCE1})  \
-      -DGITLAB_TOKEN=%(cat %{SOURCE2})  \
-     .
+%cmake -DCHUMGUI_VERSION=%(echo %{version} | grep -Eo '^[0-9]+(\.[0-9]+)*')  \
+       -DCMAKE_SKIP_RPATH:BOOL=OFF  \
+       -DCMAKE_INSTALL_RPATH=%{_datadir}/%{name}/lib:  \
+       -DGITHUB_TOKEN=%(cat %{SOURCE1})  \
+       -DGITLAB_TOKEN=%(cat %{SOURCE2})  \
+       .
 cmake --build .
 
 %install
-rm -rf %{buildroot}
 %make_install
-desktop-file-install --delete-original       \
-  --dir %{buildroot}%{_datadir}/applications \
-   %{buildroot}%{_datadir}/applications/*.desktop
+desktop-file-install --delete-original  \
+                     --dir %{buildroot}%{_datadir}/applications  \
+                     %{buildroot}%{_datadir}/applications/*.desktop
 
 %if 0%{?bundle_yaml}
 mkdir -p %{buildroot}%{_datadir}/%{name}/lib
 cp -a %{_libdir}/libyaml-cpp.so.* %{buildroot}%{_datadir}/%{name}/lib
-# strip executable bit from all libraries
+# Strip executable bit from all libraries
 chmod -x %{buildroot}%{_datadir}/%{name}/lib/*.so*
 %endif
 
-%if 0%{?sailfishos_version} && 0%{?sailfishos_version}<40100
+# Rectify desktop file for old SailfishOS releases < 4.0.1.00
+%if 0%{?sailfishos_version} && 0%{?sailfishos_version} < 40100
 sed -i 's/silica-qt5/generic/' %{buildroot}%{_datadir}/applications/sailfishos-chum-gui.desktop
 %endif
 
 %postun
-ssu rr sailfishos-chum || true
-ssu rr sailfishos-chum-testing || true
-rm -f /var/cache/ssu/features.ini || true
-ssu ur || true
+if [ $1 = 0 ]  # Removal
+then
+  ssu rr sailfishos-chum
```

Comment:
you have dropped `|| true` from several commands over here. Not sure it is 100% safe as we have to deal with the situations where users could adjust ssu repositories manually. In the worst case, we can get some issue during SFOS upgrades that would break upgrade as well. I haven't had time to think it through and check what will happen if this repository doesn't exist.

Reply:
> you have dropped `|| true` from several commands over here.

Yes, IIRC we discussed one aspect (see below) before, but as I can be the master of forgetting, I try to document my considerations (also for myself), see [comment in lines 93 - 97](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/131/files#diff-fd907f5104ba9eb63aad99adea31eeea2340cbde417219f33bba2bc4b5878257R93-R97).

> Not sure it is 100% safe as we have to deal with the situations where users could adjust ssu repositories manually.

1. Already discussed:
   *`ssu`, `rm -f`, `mkdir -p` etc. *always* return with "0" ("success"), hence no appended `|| true` needed to satisfy `set -e` for failing commands outside of flow control directives (if, while, until etc.).*
   Just try (as root): `ssu rr abcd-efgh; echo $?`
2. That was a new insight for me:
   *Furthermore on Fedora Docs it  is indicated that solely the final exit status of a whole scriptlet is crucial:  https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_syntax*
   This might be the reason why I do not see the commands scriptlets of other spec files being consistently appended with `|| true` /  `|| :`, but do concur also I experienced issues with failing commands in scriptlets: Likely they were placed in a control flow path, which (conditionally or not) let them be the last command of the scriptlet being executed, thus (occasionally) resulting a failing installation of an RPM package.

> In the worst case, we can get some issue during SFOS upgrades that would break upgrade as well.

* Generally: 
  1. Yes, packages failing to be upgraded did break SailfishOS upgrades.  But this is not applicable to this MR, as denoted above.
  2. AFAIU Jolla's efforts to isolate the repositories needed for upgrading SailfishOS since SailfishOS 3.4.0 alleviated this issue (but created new ones: [Packages becoming removed during an upgrade in an undocumented scheme](https://forum.sailfishos.org/t/how-to-interpret-uninstall-the-following-in-the-sailfishos-updater/9144/2?u=olf)).
* Specifically:
   > I haven't had time to think it through and check what will happen if this repository doesn't exist.
   
   I am pretty sure, that nothing will happen, which would not also have happened before with the `|| true`s appended.

Reply:
Thank you very much, resolving